services/app-web/src/components/SubmissionSummarySection/ContractDetailsSummarySection/ContractDetailsSummarySection.tsx

```diff
@@ -151,66 +126,84 @@ export const ContractDetailsSummarySection = ({
                     <DataDetail
                         id="federalAuthorities"
                         label="Active federal operating authority"
-                        data={createCheckboxList({
-                            list: submission.federalAuthorities,
-                            dict: FederalAuthorityRecord,
-                        })}
+                        explainMissingData={!isSubmitted}
+                        data={
+                            <DataDetailCheckboxList
+                                list={submission.federalAuthorities}
+                                dict={FederalAuthorityRecord}
+                            />
+                        }
                     />
                 </DoubleColumnGrid>
-                {submission.contractType === 'AMENDMENT' &&
-                    submission.contractAmendmentInfo && (
-                        <DoubleColumnGrid>
-                            <DataDetail
-                                id="modifiedProvisions"
-                                label="This contract action includes new or modified provisions related to the following"
-                                data={createCheckboxList({
-                                    list: modifiedProvisions,
-                                    dict: ModifiedProvisionsRecord,
-                                })}
-                            />
-
-                            <DataDetail
-                                id="unmodifiedProvisions"
-                                label="This contract action does NOT include new or modified provisions related to the following"
-                                data={createCheckboxList({
-                                    list: unmodifiedProvisions,
-                                    dict: ModifiedProvisionsRecord,
-                                })}
-                            />
-                        </DoubleColumnGrid>
-                    )}
+                {submission.contractType === 'AMENDMENT' && (
+                    <DoubleColumnGrid>
+                        <DataDetail
+                            id="modifiedProvisions"
+                            label="This contract action includes new or modified provisions related to the following"
+                            explainMissingData={!isSubmitted}
+                            data={
+                                <DataDetailCheckboxList
+                                    list={modifiedProvisions}
+                                    dict={ModifiedProvisionsRecord}
+                                />
+                            }
+                        />
+
+                        <DataDetail
+                            id="unmodifiedProvisions"
+                            label="This contract action does NOT include new or modified provisions related to the following"
+                            explainMissingData={!isSubmitted}
+                            data={
```

Comment:
I wonder if using `children` rather than a `data` prop would make all this cleaner. Might make storybook happy if your theory is right

Reply:
Yeah I was thinking about that. Thought it was a lot of code change to do for the goal of this PR but looking at things again its worth exploring at least if that would fix storybook. Would be good to know